ESPixelStick/src/ESPixelStick.h

```diff
@@ -112,6 +114,74 @@ bool setFromJSON (T& OutValue, J& Json, N Name)
     return HasBeenModified;
 };
 
+// Safer RGB to HTML string (e.g., "#FF8833") conversion function
+// Template function takes array of characters as **reference**
+// to allow template to statically assert buffer is large enough
+// to always succeed.  Returns esp_err_t to allow use of
+// ESP_ERROR_CHECK() macro (as this is used extensively elsewhere).
+template <size_t N>
+inline esp_err_t saferRgbToHtmlColorString(char (&output)[N], uint8_t r, uint8_t g, uint8_t b) {
```

Comment:
This does not belog in the Main include file. It can go into a utility include and then be included by modules that need it.